libs/common/tests/context_builder_test.cpp

```diff
@@ -176,4 +175,70 @@ TEST(ContextBuilderTests, AccessKindBuilderMultipleTimes) {
     EXPECT_TRUE(context.Get("user", "isCat").AsBool());
 }
 
+TEST(ContextBuilderTests, AddAttributeToExistingContext) {
+    auto context = ContextBuilder()
+                       .Kind("user", "potato")
+                       .Name("Bob")
+                       .Set("city", "Reno")
+                       .SetPrivate("private", "a")
+                       .Build();
+
+    auto builder = ContextBuilder(context);
+    if (auto updater = builder.Kind("user")) {
+        updater->Set("state", "Nevada");
+        updater->SetPrivate("sneaky", true);
+    }
+    auto updated_context = builder.Build();
+
+    EXPECT_EQ("potato", updated_context.Get("user", "key").AsString());
+    EXPECT_EQ("Bob", updated_context.Get("user", "name").AsString());
+    EXPECT_EQ("Reno", updated_context.Get("user", "city").AsString());
+    EXPECT_EQ("Nevada", updated_context.Get("user", "state").AsString());
+
+    EXPECT_EQ(2, updated_context.Attributes("user").PrivateAttributes().size());
+    EXPECT_EQ(1, updated_context.Attributes("user").PrivateAttributes().count("private"));
+    EXPECT_EQ(1, updated_context.Attributes("user").PrivateAttributes().count("sneaky"));
+}
+
+TEST(ContextBuilderTests, AddKindToExistingContext) {
+    auto context = ContextBuilder()
+                       .Kind("user", "potato")
+                       .Name("Bob")
+                       .Set("city", "Reno")
+                       .Build();
+
+    auto updated_context =
```

Comment:
@mmrj Adding an "org" to the context.

Reply:
nice. Do I need to call `IdentifyAsync` with `updated_context` after this?

Reply:
Yes, this just makes a context. You need to identify with that context to get flag values for it.